dashboard/test/controllers/courses_controller_test.rb

```diff
@@ -85,6 +85,16 @@ class CoursesControllerTest < ActionController::TestCase
     assert_redirected_to '/courses/csp-2018/?redirect_warning=true'
   end
 
+  test "show: do not redirect to latest stable version if no_redirect query param provided" do
+    sign_out @teacher
+    create :course, name: 'csp-2017', family_name: 'csp', version_year: '2017', is_stable: true
+    create :course, name: 'csp-2018', family_name: 'csp', version_year: '2018', is_stable: true
+
+    get :show, params: {course_name: 'csp-2017', no_redirect: "true"}
+
+    assert_response :ok
```

Comment:
My own ignorance: I had to think carefully and look some things up to understand that this checks for exactly `200 OK`, unlike `assert_response :success` which matches any 2XX, and that either would confirm that the response to this request was _not_ a (3XX) redirect. :man_facepalming: